crate/operator/main.py

```diff
@@ -592,3 +593,37 @@ async def update_cratedb_resource(
                     await ensure_user_password_label(
                         core, namespace, user_spec["password"]["secretKeyRef"]["name"]
                     )
+
+
+@kopf.on.update("", "v1", "secrets", labels={LABEL_USER_PASSWORD: "true"})
+async def secret_update(
+    namespace: str,
+    name: str,
+    diff: kopf.Diff,
+    **kwargs,
+):
+    for operation, field_path, old_value, new_value in diff:
+        if field_path == ("data", "password"):
```

Comment:
`password` isn't guaranteed to be the key in the secret that contains a password. It can be anything. The password that is used for a specific user is defined in the custom resource under `.spec.users.*.password.secretKeyRef.key`.

In essence, I think you'll need to iterate over all users in all cluster custom resources in the namespace of this secret, and check if the password for that user was changed:

```
for cluster in all_clusters_in_namespace:
    for user_spec in cluster.users:
        if user_spec.secretKeyRef.name == name_of_secret_that_was_changed and user_spec.secretKeyRef.key in diff:
            update_user_password
```

Reply:
It makes sense ... I haven't been sure about that.